syntax = "proto3";
package hb;

import "google/protobuf/empty.proto";

service HeartbeatService {
  // 3s per round to elect
  rpc Ping      (google.protobuf.Empty) returns (google.protobuf.Empty);
  // 主节点选举：返回节点启动时长
  rpc GetUptime (google.protobuf.Empty) returns (UptimeInfo);
}

message UptimeInfo {
  string node_id    = 1;   // 8-char UUID
  double uptime_sec = 2;   // 自进程启动以来的秒数
}
